ui/src/builder/BuilderFieldsText.vue

```diff
@@ -76,6 +76,16 @@ const templateField = computed(() => {
 	return definition.fields[fieldKey.value];
 });
 
+const options = computed(() => {
+	const field = templateField.value;
+	if (field.options) {
+		return typeof field.options === "function"
```

Comment:
Hey @FabienArcellier check this out

@raaymax worked on a dynamic list of options. You can pass it hardcoded options (as we do now), but you can also pass it a function to generate the options.

This was one of the challenges I saw for the _Reuse Component_ component. So with this we're halfway there. I've asked him to work on _Reuse Component_.

@raaymax for your reference https://github.com/streamsync-cloud/streamsync/issues/215

Reply:
By the way I'm talking specifically about selecting the `id` of the component that we want to reuse.

Reply:
I investigated this case, and it seems that it won't be that simple. This specific example works, but unfortunately, this approach won't work for most cases. Actually, this isn't valid code from the perspective of this framework, unfortunately. I need to investigate it further; maybe I will find a better solution for that.